src/sparseml/onnx/utils/graph_editor.py

```diff
@@ -242,6 +242,58 @@ def delete_unused_initializers(self):
             ]
         )  # delete inits that have no edge
 
+    def find_orphaned_nodes(self, node: NodeProto) -> List[NodeProto]:
+        """
+        Given a node, that is to be removed from the graph, find all nodes that
+        will be orphaned as a result of the removal. Orphaned nodes are nodes
+        that will have no inputs after the removal of the given node.
+        The method traverses the graph upwards from the given node until
+        a node with multiple outputs is found. All nodes that are traversed
+        are considered orphaned and will be removed.
```

Comment:
It seems there is a potential for this to fail an edge case where the first node with multi-outputs has all its outputs route to different part of the same orphaned subgraph i.e. the multi-output node will be orphaned as well. Is it foreseeable that this would come up in LLM graphs?

Reply:
Don't follow, could you maybe sketch the graph for that case?

Reply:
in general, nodes don't have multiple outputs unless potentially they are a graph output - but agree we should look into this edge case. @KSGulin can you sync w/ @dbogunowicz or propose a solution (this should be a simple fix if it is indeed an issue, no?)

Reply:
@KSGulin Feel free to sync with me spontaneously today or tomorrow, or otherwise, we can land this and create a small patch for this next week.